internal/router/routes.go

```diff
@@ -31,11 +31,25 @@ func SetUpRouter() *gin.Engine  {
 		exchangeFetch.GET(":stock_name/depth", controller.ViewMarketDepth)
 	}
 
-	//websocket:= r.Group("/socket")
+	//websocket := r.Group("/socket")
 	//{
 	//	websocket.GET("/", webSocket.Home)
 	//	websocket.GET("/stocks", webSocket.StockHandler)
 	//	websocket.GET("/orders", webSocket.OrderHandler)
 	//}
+	payments := r.Group("/payments")
+	{
+		payments.POST(":Userid/addAmount", controller.AddAmount)
+		payments.POST(":Userid/withdrawAmount", controller.WithdrawAmount)
+		payments.GET(":payment_status", controller.Callback)
```

Comment:
There are multiple parameters other than payment_status right? You can consume them together and process. have you checked about the payment_status param as first query param in callback?